src/cheshire/core.clj

```diff
@@ -5,12 +5,44 @@
             [cheshire.generate-seq :as gen-seq]
             [cheshire.parse :as parse])
   (:import (com.fasterxml.jackson.core JsonParser JsonFactory
-                                       JsonGenerator
+                                       JsonGenerator PrettyPrinter
                                        JsonGenerator$Feature)
            (com.fasterxml.jackson.dataformat.smile SmileFactory)
+           (cheshire.prettyprint CustomPrettyPrinter)
            (java.io StringWriter StringReader BufferedReader BufferedWriter
                     ByteArrayOutputStream OutputStream Reader Writer)))
 
+(defonce default-pretty-print-options
+  {:indentation "  "
+   :line-break "\n"
+   :indent-arrays? false
+   :indent-objects? true
+   :before-array-values nil
+   :after-array-values nil
+   :object-field-value-separator nil})
+
+(defn create-pretty-printer
+  "Returns an instance of CustomPrettyPrinter based on the configuration
+  provided as argument"
+  [options]
+  (let [{:keys [indentation
```

Comment:
Oh yeah, another thing I found when benchmarking (very sad), is that destructuring is much slower than pulling the values out of the map with keyword syntax, so perhaps this could change to let-bind the options without the destructuring?